shared/extension/Configuration.hpp

```diff
@@ -346,67 +346,82 @@ namespace NUClear::dsl {
+                    const std::string hostname = utility::support::getHostname();
+                    const std::string platform(::extension::Configuration::getPlatform(hostname));
+                    const auto binaryName = get_first_command_line_arg();
+
+                    // Get relative path to config file.
+                    const auto components = utility::strutil::split(watch.path, '/');
+                    fs::path relativePath{};
+                    bool flag = false;
+                    for (const auto& component : components) {
+                        // Ignore the hostname/binary name if they are present.
+                        if (flag && (component != hostname) && (component != binaryName) && (component != platform)) {
+                            relativePath = relativePath / component;
+                        }
+
+                        // Want paths relative to the config folder.
+                        if (component == "config") {
+                            flag = true;
                         }
-                        return std::make_shared<::extension::Configuration>(relativePath,
-                                                                            hostname,
-                                                                            binaryName,
-                                                                            platform);
-                    }
-                    catch (const YAML::ParserException& e) {
-                        throw std::runtime_error(watch.path + " " + std::string(e.what()));
                     }
+                    return std::make_shared<::extension::Configuration>(relativePath, hostname, binaryName, platform);
                 }
-                else {
-                    // Return an empty configuration (which will show up invalid)
-                    return std::shared_ptr<::extension::Configuration>(nullptr);
+                catch (const YAML::ParserException& e) {
+                    throw std::runtime_error(watch.path + " " + std::string(e.what()));
+                }
+            }
+
+            /// @brief Checks if the Configuration Reaction's parameters are valid
+            /// @details Parameters are not valid if the FileWatch object doesn't exist or it isn't a yaml file.
+            /// If this is the installation phase (denoted by the NO_OP FileWatch event), and this is not the default
+            /// configuration file, then the Reaction will also not run. This is to prevent Configuration Reactors
+            /// running multiple times during installation for one module.
+            /// @throws std::runtime_error if there is a YAML parsing error
+            /// @tparam DSL Magic NUClear type. Ignore for the purpose of understanding this function
+            /// @param t The associated Configuration Reaction
+            /// @return True if the file is valid, otherwise false.
```

Comment:
```suggestion
            /// @return False is the reaction is not to be run, otherwise true
```
Can return true or false if file is a `NO_OP`.